File: YazilimGelistiriciDotNetHomeWork/source/repos/KampHomeWorkJune/Day5Homework5GamerBackEnd/Concretes/GameSalesManager.cs
using Day5Homework5GamerBackEnd.Entities;
using System;
using System.Collections.Generic;
using System.Text;

namespace Day5Homework5GamerBackEnd.Concretes
{
    class GameSalesManager
    {
		public void purchasedGames(Gamer gamer, Game[] games)
		{
			foreach (Game game in games)
			{
				updateStockAfterPurchase(gamer, game);
			}
		}
		public void updateStockAfterPurchase(Gamer gamer, Game game)
		{
			game.AmountInStock = game.AmountInStock - 1;
			Console.WriteLine(gamer.FirstName + " bir adet oyun aldi. Stoktaki " + game.GameTitle + " oyunu sayisi : " + game.AmountInStock);
		}
	}
}

File: JavaReactKampHomeWork/homeworkJavaSE1.8June/src/day4Homework3GamerBackEnd/concretes/GameSalesManager.java
package day4Homework3GamerBackEnd.concretes;

import day4Homework3GamerBackEnd.abstracts.BaseGameManager;
import day4Homework3GamerBackEnd.entities.Game;
import day4Homework3GamerBackEnd.entities.Gamer;

public class GameSalesManager extends BaseGameManager{
	BaseGameManager baseGameManager;
	public GameSalesManager(BaseGameManager baseGameManager){
		this.baseGameManager = baseGameManager;
	}
	
	public void purchasedGames(Gamer gamer, Game[] games) {
		for(Game game : games) {
			updateStockAfterPurchase(gamer, game);
		}		
	}
	public void updateStockAfterPurchase(Gamer gamer, Game game) {
		 game.setAmountInStock(game.getAmountInStock()-1);
		 System.out.println(gamer.getFirstName() + " bir adet " + game.getGameTitle() + " oyunu aldi. Stoktaki "+ game.getGameTitle() + " oyunu sayisi : " + game.getAmountInStock());
	}

}
